modules/tactical/src/backend.ts

```diff
@@ -65,12 +103,45 @@ export class FakeBackend implements Backend {
   load(object: VersionedObject, broadcast: boolean = true): void {
     this.objects[serializeValue(object.key)] = object;
     if (broadcast) {
-      this.subject.onNext(object);
+      this.dataSubject.onNext(object);
     }
   }
 
+  mutate(key: Object, value: Object, baseVersion: string, mutationId: Object): void {
+    var keyStr = serializeValue(key);
+    // Check for version compatibility.
+    if (this.objects.hasOwnProperty(keyStr)) {
+      var old = this.objects[keyStr];
+      if (old.version !== baseVersion) {
+        // Newer version is available. Reject mutation by returning new version.
+        this.dataSubject.onNext(old);
+        return;
+      }
+    }
+    if (mutationId.hasOwnProperty('fail') && mutationId['fail']) {
```

Comment:
Keep in mind this is the FakeBackend - the testing mock. The 'fail' property is used to signal to the mock that it should pretend the request is bad, and fail it.

Eventually we will want to allow the application to attach arbitrary metadata to mutations (and persist it in the TS). This will allow the application to maintain context down the line if the mutation fails on the server.